### PR TITLE
Update sharing behavior for @ManagedExecutorConfig and add support for @NamedInstance

### PR DESCRIPTION
**Pull request by aguibert**

### CODE REVIEW

**Comment by aguibert** [comment]
#build

File: dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ConcurrencyCDIExtension.java [hunk 1]
```diff
@@ -36,13 +56,129 @@
...
+
+        // If 2 or more InjectionPoints define @NamedInstance("X") @ManagedExecutorConfig it is an error
+        if (prevoiusConfig != null) {
+            String msg = "ERROR: Found existing bean with name=" + name; // TODO NLS message
+            Tr.error(tc, msg);
+            deploymentErrors.add(new Throwable(msg));
+        }
+    }
+
+    private CDIService getCDIService() {
```
Comment (njr-11) [inline dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ConcurrencyCDIExtension.java]: Given that this class is a declarative services component, why isn't CDIService bound as a dependency?

File: dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ConcurrencyCDIExtension.java [hunk 2]
```diff
@@ -36,13 +56,129 @@
...
+        // Register 1 bean per un-named config, and 1 bean per unique NamedInstance
+        for (Entry<String, ManagedExecutorConfig> e : beanMap.get(appName).entrySet()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+                Tr.debug(tc, "Add bean for name=" + e.getKey());
+            event.addBean(new ManagedExecutorBean(e.getKey(), e.getValue()));
+        }
+    }
+
+    public void registerErrors(@Observes AfterDeploymentValidation event) {
+        deploymentErrors.forEach(event::addDeploymentProblem);
```
Comment (njr-11) [inline dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ConcurrencyCDIExtension.java]: deploymentErrors does not ever appear to be cleared.  If someone makes multiple attempts iterating over installing an application, will it keep accumulating/retaining errors from previous attempts?

**Comment by aguibert** [comment]
#build

**Pull request merged**